pandas/tests/indexing/test_coercion.py

```diff
@@ -432,13 +432,19 @@ def test_insert_index_datetimes(self, fill_val, exp_dtype):
         )
         self._assert_insert_conversion(obj, fill_val, exp, exp_dtype)
 
-        msg = "Passed item and index have different timezone"
         if fill_val.tz:
-            with pytest.raises(ValueError, match=msg):
+            msg = "Cannot compare tz-naive and tz-aware"
```

Comment:
@TomAugspurger is this what you are talking about? this I think was just wrong before, not an API change.

Reply:
Yeah. I see now that we raise TypeError for `Timestamp == Timestamp`, which I think is incorrect but not worth changing.

I still thing think this merits a release note.

Reply:
sure, @jbrockmendel can you add for what is the user viisble change.

Reply:
updated with whatsnew + green